modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -2624,6 +2627,24 @@ void TFImporter::parseActivation(tensorflow::GraphDef& net, const tensorflow::No
     connectToAllBlobs(layer_id, dstNet, parsePin(layer.input(0)), id, num_inputs);
 }
 
+void TFImporter::parseArg(tensorflow::GraphDef& net, const tensorflow::NodeDef& layer, LayerParams& layerParams)
+{
+    const std::string& name = layer.name();
+    const std::string& type = layer.op();
+
+    Mat dimension = getTensorContent(getConstBlob(layer, value_id, 1));
+    CV_Assert(dimension.rows == 1 && dimension.cols == 1);
+    CV_Assert(dimension.type() == CV_32SC1);
+    int* axis = (int*)dimension.data;
+    layerParams.set("axis", axis[0]);
```

Comment:
I think it would be better to do this instead:

```cpp

CV_Assert(dimension.total() == 1 && dimension.type() == CV_32SC1);
layerParams.set("axis", *dimension.ptr<int>());

```

That way we don't need to use `static_cast`.

Reply:
Thanks for your review, I have updated it.